devops/jobs/ImageBuilder.groovy

```diff
@@ -26,54 +26,87 @@ import static org.edx.jenkins.dsl.Constants.common_wrappers
                 }
-            }
-            logRotator common_logrotator
 
-            // add APP_NAME as parameter; will be passed in from upstream job as application Dockerfile to build
-            parameters{
-                stringParam('APP_NAME')
-            }
+                description('\rThis job builds the ' + app_name + ' Docker image and pushes it to DockerHub. ' + 
+                    'It is typically called from the upstream ' + app_name + '-watcher job or the configuration-watcher job, ' +
+                    'but it can also be run manually to build and push the ' + app_name + ' Docker image. If the application has ' +
+                    'no application code repository, it will only be triggered by changes picked up by the configuration-watcher job, or ' +
+                    'it must be built by hand.')
 
-            def access_control = extraVars.get('ACCESS_CONTROL',[])
-            access_control.each { acl ->
-                common_read_permissions.each { perm ->
-                    authorization {
-                        permission(perm,acl)
+                wrappers common_wrappers
+                // add credentials to log in to DockerHub; ID refers to credential ID as set up in Jenkins
+                wrappers {
+                    credentialsBinding {
+                        file("CONFIG_JSON_FILE", extraVars.get("CONFIG_JSON_FILE_CREDENTIAL_ID", "docker-config-json"))
+                    }
+                    // if the application has a repository, the build name will display the version of the application used; otherwise, display
+                    // only the configuration version
+                    if (app_config.get("app_repo")) {
+                        // add master commits of application repository and configuration repository that are used to build the image
+                        buildName('#${BUILD_NUMBER} with app. version = ${ENV:0:7, var="APP_VERSION"} and config. version = ${GIT_REVISION, length=8}')
+                    }
+                    else { 
+                        buildName('#${BUILD_NUMBER} with config. version = ${GIT_REVISION, length=8}')
+                    }
+                }
+                logRotator common_logrotator
+
+                // add APP_VERSION as parameter; will be passed in from upstream job as version of application being built (for build name)
+                // applications without code repositories do not have this parameter to avoid confusion when performing a manual build
+                // the default value is a string of 8 spaces; this avoids the issue of someone triggering this build manually without supplying 
+                // a value for APP_VERSION, because if the value is less than 8 spaces, the buildName is not displayed because of the substring
+                if (app_config.get("app_repo")) {
+                    parameters {
+                        stringParam('APP_VERSION', '        ', 
```

Comment:
Now that we have ecommerce-image-builder is naming the run as important?  You can just click in and see the versions on the build page.
If that helps remove a weird workaround, I'm for removing the part that names the run